RyuSocks/Auth/UsernameAndPassword.cs

```diff
@@ -25,19 +28,48 @@ namespace RyuSocks.Auth
     [AuthMethodImpl(0x02)]
     public class UsernameAndPassword : IProxyAuth
     {
+        private readonly Dictionary<string, string> _database;
+
+        public UsernameAndPassword(Dictionary<string, string> database)
+        {
+            _database = database;
+        }
+
         public bool Authenticate(ReadOnlySpan<byte> incomingPacket, out ReadOnlySpan<byte> outgoingPacket)
         {
-            throw new NotImplementedException();
+            UsernameAndPasswordRequest requestPacket = new(incomingPacket.ToArray());
+
+            requestPacket.Validate();
+
+            if (_database.TryGetValue(requestPacket.Username, out string password) &&
+                password == requestPacket.Password)
+            {
+                UsernameAndPasswordResponse successResponsePacket = new()
+                {
+                    Version = 0x01,
```

Comment:
Version seems to be a constant value. Could you define that at the top of this class as a `const`?